File: peernexus-frontend/src/services/answerService.js
import apiClient from "./apiClient.js";

export const answerService = {
  getByDoubt: (doubtId, params) =>
    apiClient.get(`/api/answers/doubt/${doubtId}`, { params }).then((r) => r.data.data),
  getById: (id) => apiClient.get(`/api/answers/${id}`).then((r) => r.data.data),
  create: (payload) => apiClient.post("/api/answers", payload).then((r) => r.data.data),
  update: (id, payload) => apiClient.put(`/api/answers/${id}`, payload).then((r) => r.data.data),
  delete: (id) => apiClient.delete(`/api/answers/${id}`).then((r) => r.data),
  accept: (id) => apiClient.post(`/api/answers/${id}/accept`).then((r) => r.data.data),
  vote: (id, type) => apiClient.post(`/api/answers/${id}/vote`, { type }).then((r) => r.data.data),
};

File: peernexus-backend/src/main/java/com/peernexus/peernexus/answer/service/AnswerService.java
package com.peernexus.peernexus.answer.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.peernexus.peernexus.answer.dto.AnswerCreateRequest;
import com.peernexus.peernexus.answer.dto.AnswerResponse;
import com.peernexus.peernexus.answer.dto.AnswerUpdateRequest;
import com.peernexus.peernexus.answer.dto.VoteRequest;

public interface AnswerService {

    AnswerResponse create(AnswerCreateRequest request);

    AnswerResponse update(Long id, AnswerUpdateRequest request);

    void delete(Long id);

    AnswerResponse getById(Long id);

    Page<AnswerResponse> getByDoubt(Long doubtId, Pageable pageable);

    AnswerResponse acceptAnswer(Long id);

    AnswerResponse vote(Long id, VoteRequest request);
}
